### PR TITLE
Update to latest changes in the plugins repository.

### PR DESCRIPTION
**Pull request by russel**
The Cargo patching still seems not to work. If we could add this changeset to see if the compilation problem goes away not using the Cargo patch, that would be very helpful.

### CODE REVIEW

File: gir-files/GstMpegts-1.0.gir [hunk 1]
```diff
@@ -108,9 +108,11 @@ Consult the relevant specifications for more details.</doc>
             c:symbol-prefix="atsc_eit">
       <doc xml:space="preserve">Event Information Table (ATSC)</doc>
       <field name="source_id" writable="1">
+        <doc xml:space="preserve">The source id</doc>
```
Comment (sdroege) [inline gir-files/GstMpegts-1.0.gir]: Should probably regenerate the gir with autotools. Meson adds a lot of docs in there and now this one is different than the other .gir files :)

File: gstreamer-mpegts-sys/Cargo.toml [hunk 2]
```diff
@@ -25,13 +25,6 @@ shell-words = "0.1.0"
 tempdir = "0.3"
 
 [features]
-v1_2 = []
-v1_4 = ["v1_2"]
-v1_6 = ["v1_4"]
-v1_8 = ["v1_6"]
-v1_10 = ["v1_8"]
-v1_12 = ["v1_10"]
-v1_14 = ["v1_12"]
```
Comment (sdroege) [inline gstreamer-mpegts-sys/Cargo.toml]: This should stay

File: gstreamer-mpegts-sys/build.rs [hunk 3]
```diff
@@ -16,22 +16,8 @@ fn main() {
...
-    } else if cfg!(feature = "v1_8") {
-        "1.8"
-    } else if cfg!(feature = "v1_6") {
-        "1.6"
-    } else if cfg!(feature = "v1_4") {
-        "1.4"
-    } else if cfg!(feature = "v1_2") {
-        "1.2"
-    } else {
-        "1.0"
```
Comment (sdroege) [inline gstreamer-mpegts-sys/build.rs]: This all should stay

File: gir-files/GstMpegts-1.0.gir (hunk 1 above)
Comment (russel) [reply gir-files/GstMpegts-1.0.gir]: Or fix the Meson build to do things as you want? I'm not sure I want to get involved in using Autotools again, especially with the Meson build available and the future of the GStreamer build.

File: gstreamer-mpegts-sys/Cargo.toml (hunk 2 above)
Comment (russel) [reply gstreamer-mpegts-sys/Cargo.toml]: I was a bit worried about these not being there but assumed it was due to a change in gstreamer-plugins-bad. However it seems another facet of the Meson build as opposed to the Autotools build.

File: gstreamer-mpegts-sys/build.rs (hunk 3 above)
Comment (russel) [reply gstreamer-mpegts-sys/build.rs]: As above. I am guessing it is directly related to the above and the Meson / Autotools issue.

**Comment by russel** [comment]
Given the issues, might it be best to try an equivalent build with Autotools and see if it solves the above problems without manual intervention, and results in the same state of the other changes?

File: gir-files/GstMpegts-1.0.gir (hunk 1 above)
Comment (sdroege) [reply gir-files/GstMpegts-1.0.gir]: I'll update that part, no worries :)

File: gstreamer-mpegts-sys/Cargo.toml (hunk 2 above)
Comment (sdroege) [reply gstreamer-mpegts-sys/Cargo.toml]: That's missing features in gir. Just revert that part

File: gstreamer-mpegts-sys/build.rs (hunk 3 above)
Comment (sdroege) [reply gstreamer-mpegts-sys/build.rs]: Same as above, simply revert

File: gstreamer-mpegts-sys/Cargo.toml (hunk 2 above)
Comment (russel) [reply gstreamer-mpegts-sys/Cargo.toml]: I'll see what I can do on the journey home this afternoon. I think it is just a question of doing an update to the pull request adding a careful meld between last commit and previous commit, so shouldn't be hard.

File: gstreamer-mpegts-sys/build.rs (hunk 3 above)
Comment (russel) [reply gstreamer-mpegts-sys/build.rs]: As above. :-)

File: gir-files/GstMpegts-1.0.gir (hunk 1 above)
Comment (russel) [reply gir-files/GstMpegts-1.0.gir]: Would you like me to remove these added lines as part of the update to the pull request as per below. If I can get a meld working, it should be straightforward.

**Comment by russel** [comment]
I believe I have made the changes requested for this pull request, do highlight any further that need making. Turned out to be quite easy using git difftool and meld!

**Comment by russel** [comment]
I have started a thread on the Rust forum to see if anyone has a fix/workaround to the problem of not being able to use patch to try everything out before creating pull requests.  https://users.rust-lang.org/t/cargo-patch-sections-doesnt-work/21730

File: gir-files/GstMpegts-1.0.gir (hunk 1 above)
Comment (sdroege) [reply gir-files/GstMpegts-1.0.gir]: If you can, sure. Otherwise I'll do that :)

Comment (russel) [reply gir-files/GstMpegts-1.0.gir]: It turned out to be trivial easy to do using meld, so hopefully it is all now done and ready to merge. Unless I missed something of course!

**Comment by sdroege** [comment]
Please also send the same change for gstreamer-rs :) Thanks!

**Comment by russel** [comment]
The changes all work as required, so we have definite progress. :-)  I shall progress finding how to use patch or replace in Cargo, people are giving some ideas. 

As far as I am aware none of the MPEG-TS changesets I have are in GStreamer_Rs. I have the same GIR file committed in my branch, but the branch is not merged into mainline.

**Comment by russel** [comment]
Two people on Twitter pushed me in the right direction, patch section has to be in the project/workspace Cargo.toml and not the subproject Cargo.toml. I am now able to continue without continual pull requests. Hopefully things will not be too big a bang when I do the final two pull requests.

**Pull request merged**